src/OrderbookCS/MatchingAlgorithm/LifoMatchingAlgorithm.cs

```
@@ -62,11 +62,7 @@ public MatchResult Match(IEnumerable<OrderbookEntry> bids, IEnumerable<Orderbook
                 // Think of refactoring this by including it elsewhere
                 var tradeResult = TradeUtilities.CreateTradeAndFills(orderToMatchBid.Current, orderToMatchAsk.Current,
```

Better, but pass in `FillQuantity` and `AllocationAlgorithm` to `Update`. Create `tradeResult` in `MathResultUpdate`'s `Update`

---

If `tradeResult` is created in `MatchResultCreator` then the conditional checks for `tradeResult.BuyFill.IsCompleteFill` and `tradeResult.SellFill.IsCompleteFill` will have to be placed inside `MatchResultCreator`. Is that the approach you are going for?

---

You'll probably need to rethink this. The problems stems from `matchResult` being created at the top of the function IMO. I don't like taking in a `matchResult` and returning a `matchResult` in `CreateMatchResult`, even though that's what I said you should do. The method is called `CreateMatchResult` so it should take in things needed to create a `MatchResult` and return that (not take it in as a parameter). The reason `MatchResult` is needed earlier is if there are no `bids` and no `asks`, it is returned. Maybe instead of returning an empty result we throw a custom `MatchException` and catch it somewhere higher up, logging the exception. 

Not sure what to do with the `tradeResult`

---

I would suggest that `tradeResult` and its conditionals stay as it is. As for `MatchResultCreator`, we can return a `MatchResult` without taking in a `MatchResult` as a parameter by creating a new instance of it in `CreateMatchResult`.

---

Are the test-suites supposed to have some failing tests? 

None of them throws an exception after implementing `MatchException`.